File: Homework/Homework Assignment 2/Codes/cakar_berk.rb
puts "INTRODUCTION TO USER-LOCATED LOOP CONTROL MECHANISMS IN RUBY"

while true
    puts "iterating once", ""
    break
end

for i in 1..10
    # next is the continue equivalent in Ruby
    next # continue to the next iteration without executing the rest of the loop body (puts statement)
    puts i
end

puts "SAMPLE 1: BREAK ------------------------------------------------------------"

# An example for break
for i in 0..10
    # When i is 3 in outer loop, break the outer loop
    # so that i become 0, 1, 2, and 3 instead of 0, 1, ..., 10
    if i == 3
        puts "in outer loop i is 3, now break out of the outer loop, such that i will not go beyond 3"
        break
    end

    puts "in outer loop, i is #{i}"

    j = 12;
    loop do
        # When j is 9 in inner loop, break the inner loop
        # so that j become 12, 11, 10, 9 instead of 12, 11, ..., -infinity
        if j == 9
            puts "  in inner loop, j is 9, now break out of the inner loop, such that j will not go minus infinity"
            break
        end
        puts "  in inner loop, j is #{j}"
        j -= 1
    end
end

puts "", "SAMPLE 2: NEXT -------------------------------------------------------------"

# An example for next
# Print a simple multiplication table
for i in 1..4
    # When the order of the row is divisible by 2, continue to the next iteration
    if i.even?
        puts "in outer loop i is even, now continue to the next iteration of the outer loop"
        next
    end
    for j in 1..4
        # When the order of the column is divisible by 3, continue to the next iteration
        # For example instead of 3 6 9 12 this will print 3 6 12
        if j == 3
            next
        end
        # Otherwise print the multiplication result
        print i*j, " "
    end
    puts
end

File: Homework/Homework Assignment 2/Codes/cakar_berk.py
print("INTRODUCTION TO USER-LOCATED LOOP CONTROL MECHANISMS IN PYTHON")

while (True):
    print("iterating once\n")
    break

for i in range(0, 30):
    continue # continue to the next iteration without executing the rest of the loop body (print statement)
    print(i)

print("SAMPLE 1: BREAK -------------------------------------------------------------")

# An example for break
odd = [1, 3, 5, 7, 9]
even = [2, 4, 6, 8]
result = []
# This nested loop form odd-even pairs and stores them in the result list
for i in odd:
    # When i is 7 in the outer loop, break out of the outer loop,
    # so that no pair will include 7 and 9
    if (i == 7):
        print("in outer loop i is 7, now break out of the outer loop, such that no pair in result will contain 7 and 9")
        break
    for j in even:
        # When j is 6 in the inner loop, break out of the inner loop,
        # so that no pair will include 6 and 8
        if (j == 6):
            print("  in inner loop j is 6, now break out of the inner loop, such that no pair in result will contain 6 and 8")
            break
        result.append('{}-{}'.format(i, j))
print("Resulting list: " + str(result))

print("\nSAMPLE 2: CONTINUE ----------------------------------------------------------")

# An example for continue
text = "Hello!World"
for i in range(0, 3):
    # Skip the second iteration so that HelloWorld will be printed twice
    if i == 1: continue
    print("i is {}: ".format(i), end='')
    for char in text:
        # Skip the exclamation mark
        if (char == '!'):
            continue
        print(char, end='')
    print("")
